### DISCUSSION TITLE
Show a public profile but hide followers, following and starred repo

### DISCUSSION THREAD

**Discussion by wzjoriv** [community/community#70126] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Right now, we are only given a binary decision on what we share on Github. Would it be possible to have some more fine control over what we share like being able to show a public profile but hide followers and starred repositories? As far as I understand, there is not an option to adjust the data shared with the public except for the option below which is all or nothing.

![image](https://github.com/community/community/assets/14988235/cc16bf81-caa1-4285-a083-7009589becba)

**Comment by Dineth-H** [comment]
Currently, the only way to adjust the data shared with the public is to use the all-or-nothing setting that you described. 

However, there are a few workarounds that you can use to achieve a finer degree of control. For example, you could **create a separate GitHub account for your personal projects** and **use a different account for your professional projects**. You could also **use a third-party service to hide your followers and starred repositories** from your public profile.

There is a **_feature request on GitHub's discussion forum_** for more fine-grained control over what information is shared on public profiles. The feature request has received a lot of support from users, so it is possible that GitHub will implement this feature in the future.

**Reply by wzjoriv** [reply]
I understand, but even for a personal GitHub account I would like to choose what I share with others. 

Would you mind sharing with me the discussion forum. I tried looking for one but couldn't find any so that is why I made this post.

**Reply by Dineth-H** [reply]
Found a few discussions through the community but they are currently closed. It is a good thing to start this discussion, you can try to raise your concerns:

-  #415 [here](https://github.com/AntonOsika/gpt-engineer/issues/415)
-  #773 [here](https://github.com/isaacs/github/issues/773)

some GitHub Documentations as well to refer:

- [Permissions required for fine-grained personal access tokens](https://docs.github.com/en/rest/overview/permissions-required-for-fine-grained-personal-access-tokens?apiVersion=2022-11-28)
- [Limiting interactions in your repository](https://docs.github.com/en/communities/moderating-comments-and-conversations/limiting-interactions-in-your-repository)

will edit this reply if i find another!

**Comment by najibullahjafari** [comment]
You can make individual repositories public or private. If you have repositories that you want to keep private, mark them as private in your repository settings. You can choose which repositories are visible on your profile.

**Reply by wzjoriv** [reply]
I am referring to is choosing what to share or not. For example, the people that I follow, those who follow me, projects I star, and/or my activities. I would like more fine control over what I share in general with the public/friends

**Reply by benjazzgo** [reply]
I think what OP means here is not that, @najibullahjafari. The issue here is about GitHub profile activity privacy and not repository activity.

What you are referring to indeed will apply to all activity related to a given repo in case you have control of that repo. So if I own repo X and I want the contributions on that repo, whether I starred it or not, or any other activity related to it to be private, this would be solved by making the repo private, it is correct.

But what OP is referring to goes beyond that. Currently, we have a bunch of profile-related activities:

- Activity feed on your GitHub homepage when you're logged in where you can see activity from users you follow, and therefore your followers can see your activity.
- Contribution graph where public repos contributions made by you are always visible
- Your list of followers and the list of users you follow
- Stars you gave to public repos

Now let's say you don't want to make your followers and the users you follow public. Your only alternative is to make your profile private. But by doing so, all the other items are now private too. That's the issue.

What is being proposed is to allow fine control. Something like:
- My activity on the feed of my followers - PUBLIC
- Contribution graph - PUBLIC
- My list of followers/following - PRIVATE
- List of public repos starred by me - PRIVATE

@wzjoriv please correct me if I'm wrong

**Reply by wzjoriv** [reply]
Yeah. That's what I mean

**Comment by benjazzgo** [comment]
I think it makes total sense. Particularly with followers/following list, as it can easily indirectly reveal information about who you know personally or where you work/live, even if you didn't directly provide this in your profile.

So I can see why one would want to hide that but keep all the other public profile features turned on, like the contribution graph for example.

**Comment by Luisao-official** [comment]
Not only that, but stalkers could also map the developers from a company and make targeted phishing attacks, social engineering schemes, etc.  Not to mention that it makes it easier to track and harass devs from popular projects.

Leaving starred repos to the public, exposes your personal interests and activities to potential stalkers, which may be unacceptable for some.

 This lack of fine control on our privacy options means that we can only choose between being too exposed or too strict,
 
 And the fix for that is quite simple: give options to hide certain tabs/nodes and specific data. Easy as that.

**Comment by ryuji-kimura-24** [comment]
+1

Fully agree with @wzjoriv, @benjazzgo and @Luisao-official, hopefully we will see it rolled out asap

**Comment by robbycbennett** [comment]
I would like to have my pinned repositories and nothing else on my profile.

**Comment by Alex7k** [comment]
+1
I desperately need this! Bump

**Comment by niravkukadiya** [comment]
Users should **decide** what’s visible and what’s not. Let’s push for better, more customizable privacy options.

**Comment by Alex7k** [comment]
bump
